unitylibs/utils/StringUtils.js

```diff
@@ -0,0 +1,9 @@
+export function getExtension(name) {
+    return name && name.includes('.') ? name.split('.').pop() : '';
+  }
+
+export function withoutExtension(name) {
+  if (name == null) return name;
+  const lastDot = name.lastIndexOf('.');
+  return lastDot >= 0 && lastDot < name.length - 1 ? name.substring(0, lastDot) : name;
```

Comment:
substring already handles your second check.
```suggestion
  return lastDot > 0 ? name.substring(0, lastDot) : name;
```
Please validate the 2 methods work for edge cases like empty filename with only file extension ".pdf".
Also, would advise to add string check for type safety.